### PR TITLE
Add Hebrew Months test

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
depends on https://github.com/dotnet/coreclr/pull/16247

**Comment by tarekgh** [comment]
CC @jkotas

**Comment by jkotas** [comment]
Nit: Typo in the commit description

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs [hunk 1]
```diff
@@ -131,7 +131,35 @@ public void ShortestDayNamesTest(string cultureName)
...
+            CultureInfo ci = new CultureInfo("he-IL");
+            ci.DateTimeFormat.Calendar = new HebrewCalendar();
+
+            Assert.Equal(13, ci.DateTimeFormat.MonthNames.Length);
+            Assert.Equal(13, ci.DateTimeFormat.MonthGenitiveNames.Length);
+            Assert.Equal(13, ci.DateTimeFormat.AbbreviatedMonthNames.Length);
+            Assert.Equal(13, ci.DateTimeFormat.AbbreviatedMonthGenitiveNames.Length);
+
+            DateTime dt = ci.DateTimeFormat.Calendar.ToDateTime(5779, 1, 1, 0, 0, 0, 0); // leap year
+            for (int i=0; i<13; i++)
```
Comment (krwq) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: nit: spaces

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs [hunk 2]
```diff
@@ -131,7 +131,35 @@ public void ShortestDayNamesTest(string cultureName)
             for (DayOfWeek day=DayOfWeek.Sunday; day <= DayOfWeek.Saturday; day++)
             {
                 Assert.Equal(shortestDayNames[(int) day], dtfi.GetShortestDayName(day));
-            } 
+            }
+        }
+
+        [Fact]
+        public void TestHebrewMonths()
```
Comment (krwq) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: nit: optionally I'd rename it to something with RoundTrip in the name

Comment (tarekgh) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: I am not testing round tripping more than ensuring we can parse back the formatted string.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: ok will fix the the spaces

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs (hunk 2 above)
Comment (krwq) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: Isn't that what round-tripping means? (I always thought of it as testing `X=F(F^(-1)(X))`)

Comment (tarekgh) [reply src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoTests.cs]: The test mainly interested in the Hebrew months handling, getting it from DTFI, formatting it and parse it. it doesn't matter if we are doing round tripping in here. I want to have the test name reflect what is the intention of what we are trying to test.

**Comment by tarekgh** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by tarekgh** [comment]
@dotnet-bot test this please

**Pull request merged**